mods/ra/maps/negotiations/negotiations-reinforcements.lua

```diff
@@ -0,0 +1,409 @@
+		local guard = Actor.Create("e1", false, { Owner = USSR, Location = GuardHouseSpawn.Location, SubCell = 4, Facing = Angle.SouthWest } )
+
+		Trigger.OnAddedToWorld(guard, function()
+			guard.Move(GuardHouseExit.Location)
+			guard.AttackMove(goal)
+			IdleHunt(guard)
+		end)
+
+		Trigger.AfterDelay(i * 5, function()
+			guard.IsInWorld = true
+		end)
+	end
+end
+
+ReinforceHardTeams = function()
+	if not (Difficulty == "hard") then
+		return
+	end
+
+	ReinforceBaseDefenders()
+	ReinforcePrisonPatrol()
+	ReinforceSamRocket()
+	ReinforceNorthEdgeGuards()
+	PrepareSouthChinook()
+	ReinforceHardDogs()
+end
+
+ReinforceBaseDefenders = function()
+	local types = { "3tnk", "3tnk" }
+	local defenders = Reinforcements.Reinforce(USSR, types, { BadGuyRally.Location + CVec.New(2, 2) }, 0, function(actor)
+		actor.Wait(1)
+		actor.Scatter()
+	end)
+	GroupHuntOnDamaged(defenders, Greece)
+
+	local structures = { RoadTurretWest, RoadTurretEast, ForwardPower, ForwardCommand, ForwardTech }
+	Trigger.OnAnyKilled(structures, function()
+		GroupIdleHunt(defenders)
+	end)
+end
+
+ReinforcePrisonPatrol = function()
+	local types = { "e1", "e1", "e2", "e1" }
+	local patrolPath = { VillageNortheast.Location, VillageSouthwest.Location }
+	local soldiers = Reinforcements.Reinforce(USSR, types, { NorthFlare.Location }, 0, function(actor)
+		actor.Scatter()
+	end)
+
+	Trigger.OnEnteredFootprint( { GeneralRally.Location }, function(actor, id)
+		if not (actor.Type == "gnrl") then
+			return
+		end
+
+		Trigger.RemoveFootprintTrigger(id)
+		GroupHuntOnDamaged(soldiers, Greece)
+
+		Utils.Do(soldiers, function(soldier)
+			soldier.AttackMove(BadGuyRally.Location)
+			soldier.AttackMove(GuideBarrelGoal.Location)
+			soldier.AttackMove(VillageNorthwest.Location)
+			soldier.Patrol(patrolPath, true)
+		end)
+	end)
+end
+
+ReinforceSamRocket = function()
+	local rocket = Actor.Create("v2rl", true, { Owner = USSR, Facing = Angle.West, Location = SamRocketEntry.Location } )
+	Reinforcements.Reinforce(USSR, { "e1" }, { SamFenceReveal.Location }, 0, function(actor)
+		actor.Guard(rocket)
+	end)
+end
+
+PrepareSouthChinook = function()
+	local startPosition = SouthChinookEntry.CenterPosition + WVec.New(0, 0, Actor.CruiseAltitude("tran"))
+	local transport = Actor.Create("tran.south", false, { Owner = USSR, CenterPosition = startPosition, Facing = Angle.East } )
+
+	Trigger.OnAddedToWorld(transport, function()
+		transport.UnloadPassengers(SouthChinookEntry.Location)
+		transport.Move(SouthChinookExit.Location)
+		transport.Destroy()
+	end)
+
+	Utils.Do(transport.Passengers, function(passenger)
+		Trigger.OnAddedToWorld(passenger, function()
+			passenger.Wait(DateTime.Seconds(1))
+			IdleHunt(passenger)
+		end)
+	end)
+
+	Trigger.OnEnteredProximityTrigger(SouthChinookProximity.CenterPosition, WDist.FromCells(8), function(actor, id)
+		if not (EscortFinished and actor.Type == TanyaType) then
+			return
+		end
+
+		Trigger.RemoveProximityTrigger(id)
+		transport.IsInWorld = true
```

Comment:
Again, can we `Actor.Create` the unit here instead of adding things manually to the world?

Reply:
Will do for this and the others. For this one, I believe I was trying to get the creation done and out of the way by the mission's start instead of doing it mid-game. I think it's less readable as-is though.